README.dev.md

```diff
@@ -6,11 +6,612 @@ You will need a machine with at least 2GB of RAM available (see
 https://github.com/zulip/zulip/issues/32 for a plan for how to
 dramatically reduce this requirement).
 
-Start by cloning this repository: `git clone https://github.com/zulip/zulip.git`
+Contents:
+
+* [First-time contributors start here](#first-time-contributors-start-here)
+* [Using Vagrant](#using-vagrant)
+* [Using provision.py without Vagrant](#using-provisionpy-without-vagrant)
+* [By hand](#by-hand)
```

Comment:
This probably should be renamed to "Installing by hand on UNIX platforms (Debian/Ubuntu, CentOS, Fedora, OpenBSD).".  I think if we move the `git clone` instructions from the "by hand" section into the individual subsections, each individual platform here can be its own link.